src/masternodes/mn_checks.cpp

```diff
@@ -1812,6 +1819,23 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor
         return mnview.EraseLoanScheme(obj.identifier);
     }
 
+    Res operator()(const CVaultMessage& obj) const {
+        // Check LoanScheme exists
+        auto vault = CVault();
+        static_cast<CVaultMessage&>(vault) = obj;
```

Comment:
So you don't need the explicit constructor here https://github.com/DeFiCh/ain/pull/640/commits/27a463e69e9fb39d28d8eb853c6761e4dbd66572#diff-60cfde7d4f4a598c7be62e085578fbfc2efd2217e3eee81e9e9211bb9aa69072R140